### DISCUSSION TITLE
null체크 질문 드립니다 (유니티)

### DISCUSSION THREAD

**Discussion by koldshine** [naverz/zepeto-studio-kor#814] [5. Q&A]
안녕하세요

아래처럼 프로퍼티가 선언이 되어 있고 인스펙터에서 GameObject 등록은 하지 않았습니다
public testObj : GameObject

console.log(testObj)
로그 찍어보면 null로 나오는데요 아래와 같이 if문을 쓸경우 if안쪽을 탑니다
if(this.testObj){
}

breakpoint찍어서 마우스커서 올려보면 GameObject는 아니지만 null도 아니던데

이럴경우 null체크를 어떻게 해야 하는지 궁금합니다.
팁 좀 알려주세요 ㅠ

**Comment by Cha-SeongHun** [comment]
GameObject를 Destroy함수로 파괴한 뒤 해당객체를 로그 찍어보면 null이라고 출력되는데
값은 존재하는 이상한 상태가 되더라구요.

저는 아래와 같이 유틸리티 함수를 만들어 Null체크를 하고 있습니다.

```ts
// GameObject가 진짜 null인지를 검사한다...
public static IsRealNull(value: GameObject): bool {
    if(!value) return true;

    const valueString = value.to_string();
    return valueString === "null" || valueString === "undefined";
}
```

**Reply by koldshine** [reply]
감사합니다!

**Reply by chlguswls32** [reply]
감사합니다!
